src/ctl.c

```diff
@@ -3549,3 +3551,34 @@ experimental_prof_recent_alloc_max_ctl(tsd_t *tsd, const size_t *mib,
 label_return:
 	return ret;
 }
+
+typedef struct write_cb_packet_s write_cb_packet_t;
+struct write_cb_packet_s {
+	void (*write_cb)(void *, const char *);
+	void *cbopaque;
+};
+
+static int
+experimental_prof_recent_alloc_dump_ctl(tsd_t *tsd, const size_t *mib,
+    size_t miblen, void *oldp, size_t *oldlenp, void *newp, size_t newlen) {
+	int ret;
+
+	if (!(config_prof && opt_prof)) {
+		ret = ENOENT;
+		goto label_return;
+	}
+
+	assert(sizeof(write_cb_packet_t) == sizeof(size_t) * 2);
```

Comment:
Why is this required?

Reply:
It's similar to what we had for the memory utilization APIs. See https://github.com/jemalloc/jemalloc/blob/7014f81e172290466e1a28118b622519bbbed2b0/src/ctl.c#L3442.

The input structure is not public, so the caller would have to make up a `sizeof(size_t) * 2` input array, but internally I'm using this `write_cb_packet_t` structure, so I'd better assert that the two are equivalent.

Reply:
Here probably better to say `== sizeof(void *) * 2`

Reply:
Oh I see. Yes, you're right.